torch_geometric/nn/aggr/pt_utils.py

```diff
@@ -0,0 +1,115 @@
+from typing import Optional
+
+import numpy as np
+import torch
+from torch import Tensor
+
+from torch_geometric.nn.aggr.utils import MultiheadAttentionBlock
+from torch_geometric.nn.encoding import TemporalEncoding
+
+
+class FeatEncoder(torch.nn.Module):
+    r"""Returns embedded edge features.
+
+    Args:
+        feat_dim (int): edge feature dimension.
+        out_dim (int): output dimension.
+        time_dim (int, optional): time embedding dimension.
+    """
+    def __init__(
+        self,
+        feat_dim: int = 16,
+        out_dim: int = 16,
+        time_dim: int = 0,
+    ):
+        super().__init__()
+
+        self.feat_dim = feat_dim
+        self.out_dim = out_dim
+        self.time_dim = time_dim
+        if self.time_dim > 0:
+            self.time_encoder = TemporalEncoding(self.time_dim)
+        self.feat_encoder = torch.nn.Linear(self.time_dim + self.feat_dim,
+                                            self.out_dim)
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        if (self.time_dim > 0):
+            self.time_encoder.reset_parameters()
+        self.feat_encoder.reset_parameters()
+
+    def forward(self, edge_feats: Tensor, edge_ts: Tensor = None):
+        if (self.time_dim > 0):
+            edge_time_feats = self.time_encoder(edge_ts)
+            x = torch.cat([edge_feats, edge_time_feats], dim=1)
+        else:
+            x = edge_feats
+        return self.feat_encoder(x)
+
+
+class TransformerBlock(torch.nn.Module):
+    r"""transformer block for patch transformer.
+
+    Args:
+        dims (int): hidden dimension
+        heads (int): number of attention heads
+        dropout (float, optional): dropout value.
+    """
+    def __init__(self, dims: int, heads: int, dropout: float = 0.2):
+        super().__init__()
+
+        self.dims = dims
+        self.transformer_encoder = MultiheadAttentionBlock(
+            dims, heads, layer_norm=True, dropout=dropout)
+
+    def reset_parameters(self):
+        self.transformer_encoder.reset_parameters()
+
+    def forward(self, x, mask: Optional[Tensor] = None):
+        x = x + self.transformer_encoder(x, x, mask, mask)
+        return x
+
+
+class PositionalEncoding1D(torch.nn.Module):
```

Comment:
Any difference between this and the https://github.com/pyg-team/pytorch_geometric/blob/fbafbc4fc9181e8759ec1f39d9618992793b5fe1/torch_geometric/nn/encoding.py#L7 Seems that the output tensor is 3D instead of 2D?
Can we use existing one and repeat to 3D?

Reply:
I tried both and seem to get better performance with the `PositionalEncoding1D`, though they are very close